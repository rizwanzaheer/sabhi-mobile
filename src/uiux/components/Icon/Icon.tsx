import * as React from 'react'
import { withTheme } from '../../theming/index'

import Feather from 'react-native-vector-icons/Feather'
import FontAwesome from 'react-native-vector-icons/FontAwesome'
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'
import AntDesign from 'react-native-vector-icons/AntDesign'
import Entypo from 'react-native-vector-icons/Entypo'
import EvilIcons from 'react-native-vector-icons/EvilIcons'
import FontAwesome5Pro from 'react-native-vector-icons/FontAwesome5Pro'
import Foundation from 'react-native-vector-icons/Foundation'
import Ionicons from 'react-native-vector-icons/Ionicons'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'
import MaterialIcons from 'react-native-vector-icons/MaterialIcons'

const IconsFontFamilies: { [index: string]: any } = {
  Feather,
  FontAwesome,
  FontAwesome5,
  AntDesign,
  Entypo,
  EvilIcons,
  FontAwesome5Pro,
  Foundation,
  Ionicons,
  MaterialCommunityIcons,
  MaterialIcons,
}

export interface IconConfig {
  name: string

  /**
   *  'Feather' | 'FontAwesome' | 'FontAwesome5' | 'AntDesign' | 'Entypo' | 'EvilIcons' | 'FontAwesome5Pro' | 'Foundation' | 'Ionicons' | 'MaterialCommunityIcons'| 'MaterialIcons'
   */
  iconFamily: string
}

interface IconProps {
  /**
   * An icon object that contains the key name and font family. This would usually be part of a global icons objects
   */
  icon: IconConfig

  /**
   * The size of the icon. Default to 26
   */
  size?: number

  /**
   * The color of the icon. Defaults to primary theme accessory
   */
  color?: string

  /**
   * Used by the theme provider
   */
  theme: any
}

const Icon: React.FC<IconProps> = props => {
  const IconFont = IconsFontFamilies[props.icon.iconFamily]

  return (
    <IconFont
      name={props.icon.name}
      size={props.size ? props.size : 26}
      color={props.color ? props.color : props.theme.colors.primary.accessories}
    />
  )
}

export default withTheme(Icon)
