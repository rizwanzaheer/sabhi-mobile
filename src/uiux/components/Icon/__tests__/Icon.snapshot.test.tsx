import * as React from 'react'
import { render } from 'react-native-testing-library'
import Icon from '../Icon'

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon')

describe('Component(snapshots): Icon', () => {
  it('should render with required props', () => {
    const ICON = {
      name: 'ios-alert',
      iconFamily: 'Ionicons',
    }
    const tree = render(<Icon icon={ICON} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render with all props', () => {
    const ICON = {
      name: 'ios-alert',
      iconFamily: 'Ionicons',
    }
    const tree = render(<Icon icon={ICON} size={20} color={'#FAFAFA'} />)

    expect(tree).toMatchSnapshot()
  })
})
