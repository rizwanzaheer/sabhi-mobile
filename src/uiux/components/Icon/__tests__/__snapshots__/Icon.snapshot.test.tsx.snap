// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshots): Icon should render with all props 1`] = `
<Icon
  color="#FAFAFA"
  name="ios-alert"
  size={20}
/>
`;

exports[`Component(snapshots): Icon should render with required props 1`] = `
<Icon
  color="#BBBBBB"
  name="ios-alert"
  size={26}
/>
`;
